operator deployment-validation-operator (0.3.0)

npecka opened the pull request:
Updating DVO from 0.2.2 to 0.3.0

### Updates to existing Operators

* [x] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md)?
* [x] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`?
* [x] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#channels) defined in the `package.yaml` or `annotations.yaml` ?
* [x] Have you tested an update to your Operator when deployed via OLM?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?

npecka commented:
/retest-required

framework-automation commented:
/merge possible

J0zi commented:
/retest

mvalarh commented on operators/deployment-validation-operator/0.3.0/manifests/deploymentvalidationoperator.0.3.0.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,158 @@
+    matchLabels:
+      alm-owner-dvo: deployment-validation-operator
+      operated-by: deployment-validation-operator
+  version: 0.3.0
```
Can you add `replaces: deployment-validation-operator.v0.2.2`

mvalarh replied:
`      replaces: deployment-validation-operator.v0.2.2` should be in `spec` level. You can put it after `version: 0.3.0`

J0zi replied:
@npecka 
`replaces` should be as `spec.replaces`, the same level as `spec.version`
```
spec:
  ...
  replaces: deployment-validation-operator.v0.2.2
  version: 0.3.0
```

framework-automation commented:
/merge possible

mvalarh commented:
/retest

mvalarh commented:
/test 4.7-deploy-operator-on-openshift

framework-automation commented:
/merge possible

J0zi commented:
@npecka any update on this? https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/1464#discussion_r938582223

npecka commented:
@J0zi @mvalarh - Question about the PR. If we want to have a service object deployed when the operator is installed via operator hub, is the way I've done it correct? I've added the service deployment in the manifests file and added a container port to the clusterserviceversion file. Is this correct or is there more that needs to be done?

framework-automation commented:
/merge possible

J0zi commented:
> @J0zi @mvalarh - Question about the PR. If we want to have a service object deployed when the operator is installed via operator hub, is the way I've done it correct? I've added the service deployment in the manifests file and added a container port to the clusterserviceversion file. Is this correct or is there more that needs to be done?

We are checking if installation and uninstallation passed. If functionality is as expected, then this is the correct way to do so.

J0zi commented:
@npecka please move `replaces` to the correct position as `spec.replaces`

npecka commented:
> @npecka please move `replaces` to the correct position as `spec.replaces`

As a follow up to this, is using `replaces` the appropriate option or should we be using `skipRanges` instead?

framework-automation commented:
/merge possible

framework-automation commented:
/merge possible
